Apps/AccountDataAccess/src/Patient/PatientRepository.cs

```diff
@@ -120,16 +130,23 @@ public async Task BlockAccess(BlockAccessCommand command, CancellationToken ct =
 
             HashSet<DataSource> sources = command.DataSources.ToHashSet();
 
+            // commit to the database if change feed is disabled; if change feed enabled, commit will happen when message sender is called
+            // this is temporary and will be changed when we introduce a proper unit of work to manage transactions.
             if (sources.Any())
             {
                 blockedAccess.DataSources = sources;
                 blockedAccess.CreatedDateTime = DateTime.UtcNow;
                 blockedAccess.UpdatedDateTime = DateTime.UtcNow;
-                await this.blockedAccessDelegate.UpdateBlockedAccessAsync(blockedAccess, agentAudit).ConfigureAwait(true);
+                await this.blockedAccessDelegate.UpdateBlockedAccessAsync(blockedAccess, agentAudit, !this.changeFeedEnabled);
             }
             else
             {
-                await this.blockedAccessDelegate.DeleteBlockedAccessAsync(blockedAccess, agentAudit);
+                await this.blockedAccessDelegate.DeleteBlockedAccessAsync(blockedAccess, agentAudit, !this.changeFeedEnabled);
+            }
+
+            if (this.changeFeedEnabled)
+            {
+                await this.messageSender.SendAsync(new[] { new MessageEnvelope(new DataSourcesBlockedEvent(command.Hdid, blockedAccess.DataSources), command.Hdid) }, ct);
```

Comment:
I'd recommend not reusing DataSource enum in the event, so either use a string in the event or replicate the enum.

Reply:
Changed.